RebuildWithLocalMSBuild.cmd

```diff
@@ -1,13 +1,68 @@
 :: This script will:
-::   1) Rebuild MSBuild source tree.
+::   1) Build MSBuild source tree.
 ::   2) Create a copy of the build output in bin\MSBuild
 ::   3) Build the source tree again with the MSBuild.exe in step 2.
 
 @echo off
 setlocal
 
-set MSBUILD_CUSTOM_PATH=%~dp0Tools\MSBuild.exe
+:: Restore build tools
+call %~dp0init-tools.cmd
+
 set RUNTIME_HOST=%~dp0Tools\CoreRun.exe
 
-:: Build and run tests on CoreCLR
-"%~dp0build.cmd" /t:RebuildAndTest /p:Configuration=Debug-NetCore
+:: build MSBuild with the MSBuild binaries from BuildTools
+
+set MSBUILDLOGPATH=%~dp0msbuild_bootstrap.log
+set MSBUILD_CUSTOM_PATH=%~dp0Tools\MSBuild.exe
+
+call "%~dp0build.cmd" /t:RebuildAndTest /p:Configuration=Debug-NetCore
+
+if %ERRORLEVEL% NEQ 0 (
+    echo.
+    echo Bootstrap build failed with errorlevel %ERRORLEVEL% 1>&2
+    goto :error
+)
+
+:: Kill Roslyn, which may have handles open to files we want
+taskkill /F /IM vbcscompiler.exe
+
+:: Move initial build to bootstrap directory
+set MSBUILDLOGPATH=%~dp0msbuild_move_bootstrap.log
+set MSBUILD_ARGS=/v:m BootStrapMSbuild.proj /p:Configuration=Debug-NetCore
```

Comment:
I always prefer long-form arguments in scripts. Easier for humans to read, no harder for machines to understand. Short-form arguments are for when I'm lazily typing :)